### ISSUE TITLE
Improve Bio.PDB.mmcifio MMCIFIO to preserve original PDB chain ID

### ISSUE THREAD

**Issue by data2code** [biopython/biopython#4342]
### Setup
Currently, when MMCIFIO saves a structure to .cif, it rename chain IDs into A,B,C ...
This renaming causes issues with other packages, such as crashing the logic in AlphaFold.

A patch idea was described on https://github.com/sokrypton/ColabFold/issues/449, but it will make more sense to patch MMCIFIO directly to preserve the original chain ID (it should do so conceptually).

Thank you for your kind consideration.

[3fn0.zip](https://github.com/biopython/biopython/files/11897454/3fn0.zip)

### Actual behaviour

An example below converts 3fn0.pdb (attached) into 3fno.cif, but the chains are renamed.

```
from Bio.PDB import MMCIFParser, PDBParser, MMCIFIO
def convert_pdb_to_mmcif(pdb_file):
    """convert existing pdb files into mmcif with the required poly_seq and revision_date"""
    i = "mpdb"
    cif_file="out.cif"
    parser = PDBParser(QUIET=True)
    structure = parser.get_structure(i, pdb_file)
    cif_io = MMCIFIO()
    cif_io.set_structure(structure)
    cif_io.save(str(cif_file))

convert_pdb_to_mmcif("3fn0.pdb")
```

**Comment by speleo3** [comment]
In my understanding, MMCIFIO is doing the right thing. The PDB chain id is preserved as `auth_asym_id`, this is consistent with .pdb and .cif downloads from rcsb.org.

E.g. `chain=H` from https://files.rcsb.org/download/1HZH.pdb is `auth_asym_id=H` and `label_asym_id=A` in https://files.rcsb.org/download/1HZH.cif

**Comment by jkosinski** [comment]
The problem is this call:

```
label_asym_id = self._get_label_asym_id(entity_id)
```

of this function:

```
    def _get_label_asym_id(self, entity_id):
        # Convert a positive integer into a chain ID
        # Goes A to Z, then AA to ZA, AB to ZB etc
        # This is in line with existing mmCIF files
        div = entity_id
        out = ""
        while div > 0:
            mod = (div - 1) % 26
            out += chr(65 + mod)
            div = int((div - mod) / 26)
        return out
```

It starts naming from A. So if label_asym_id were B, C, D in the original file, they are A, B, C in the converted file.

**Comment by speleo3** [comment]
Which "original file"? The `pdb_file`? It does not contain a `label_asym_id` equivalent.

**Comment by jkosinski** [comment]
Yes, in the case of the PDB the label_asym_id could take chain id. At least, this is what ColabFold/AlphaFold with custom templates expect. I think some molecular viewers would also use label_asym_id so maybe just putting chain id is the safest.

**Comment by speleo3** [comment]
Quote from https://github.com/deepmind/alphafold/issues/788#issuecomment-1612902410

> the `_struct_asym` table is corrupt as it contains author chain IDs instead of internal chain IDs.

As far as I can tell, the bug should be fixed here: https://github.com/sokrypton/ColabFold/blob/e9dc2d55230c36753812b18ee64bf405ea576643/colabfold/utils.py#L195-L202
